steps/test/unit/tReorderMSData.cc

```diff
@@ -1,94 +1,115 @@
+  std::ifstream input_file_ptr(input_filename);
+
+  if (reference_file_ptr.fail() || input_file_ptr.fail()) {
     std::cout << "Failed to open file" << std::endl;
+    BOOST_CHECK(false);
   }
 
-  const std::ifstream::pos_type fileSize = file1.tellg();
-  if (fileSize != file2.tellg()) {
-    return false;  // different file size
+  // Meta files contain path information in the first line of the binary.
+  // Since the path representation differs (i.e absolute path vs relative path),
+  // we skip the path check and compare only the rest of the meta information.
+  if (isMetaFile) {
+    reference_file_ptr.ignore(std::numeric_limits<std::streamsize>::max(),
+                              '\n');
+    input_file_ptr.ignore(std::numeric_limits<std::streamsize>::max(), '\n');
+
+    if (reference_file_ptr.eof() || input_file_ptr.eof()) {
+      std::cout << "End of file reached" << std::endl;
+      BOOST_CHECK(false);
+    }
+  } else {
+    // If file sizes are unequal return false.
+    if (reference_file_ptr.tellg() != input_file_ptr.tellg()) {
+      std::cout << "Unequal binary file sizes" << std::endl;
+      BOOST_CHECK(false);
+    }
+
+    // Resetting file pointers to compare from the beginning.
+    reference_file_ptr.seekg(0);
+    input_file_ptr.seekg(0);
   }
 
-  file1.seekg(0);  // rewind
-  file2.seekg(0);  // rewind
-
-  std::istreambuf_iterator<char> begin1(file1);
-  std::istreambuf_iterator<char> begin2(file2);
+  // Setting the file stream to not to skip whitespaces.
+  reference_file_ptr >> std::noskipws;
+  input_file_ptr >> std::noskipws;
+  std::istreambuf_iterator<char> reference_file_iterator(reference_file_ptr);
+  std::istreambuf_iterator<char> input_file_iterator(input_file_ptr);
 
-  return std::equal(begin1, std::istreambuf_iterator<char>(),
-                    begin2);  // Second argument is end-of-range iterator
+  std::istreambuf_iterator<char> end_of_stream;
+  BOOST_REQUIRE_EQUAL_COLLECTIONS(reference_file_iterator, end_of_stream,
+                                  input_file_iterator, end_of_stream);
 }
 
-// Test simple averaging without flagged points.
-void test1(std::string msPath, std::string msOutPath) {
+// Test reorder functionality with default settings
+void ReorderTest(std::string ms_input_path, std::string ms_output_path) {
   // Create the steps.
   ParameterSet parset;
-  parset.add("msin", msPath);
+  parset.add("msin", ms_input_path);
   parset.add("msin.ntimes", "2");
 
-  if (msOutPath.empty()) {
-    std::string tmpPath(msPath);
-    tmpPath.resize(tmpPath.size() - 3);
-    parset.add("msout", tmpPath + "_output.ms");
+  if (ms_output_path.empty()) {
+    std::string temp_path(ms_input_path);
+    temp_path.resize(temp_path.size() - 3);
+    parset.add("msout", temp_path + "_output.ms");
   } else
-    parset.add("msout", msOutPath);
+    parset.add("msout", ms_output_path);
 
+  // Creating steps
   std::shared_ptr<dp3::steps::InputStep> input_step =
```

Comment:
For the sake of consistency this should probably also be `auto` to match the two variables below or alternatively none of them should be `auto`

Reply:
removed auto-type.